xpdo/xpdo.class.php

```diff
@@ -2034,12 +2034,31 @@ protected function _log($level, $msg, $target= '', $def= '', $file= '', $line= '
             if (!empty ($line)) {
                 $line= " : {$line}";
             }
+            switch (gettype($msg)) {
+                case 'array' :
+                    $log = print_r($msg, true);
+                    break;
+                case 'object' :
+                    if ($msg instanceof xPDOObject) {
+                        $log = print_r($msg->toArray(), true);
+                    } elseif ($msg instanceof xPDOQuery) {
```

Comment:
I wonder if we should `clone` the `$msg` if it is an xPDOQuery since we are preparing and turning it to SQL manually. Not sure the effects, but seems like it could potentially cause a problem if used before execution of the query.